src/Apps/Order/Components/AddressModal.tsx

```diff
@@ -76,21 +73,24 @@ export const AddressModal: React.FC<Props> = ({
   modalDetails,
   me,
 }) => {
-  const [_countryCode, setCountryCode] = useState<string>("us")
-
   const title = modalDetails?.addressModalTitle
   const createMutation =
     modalDetails?.addressModalAction === "createUserAddress"
+
   const validator = (values: any) => {
     const validationResult = validateAddress(values)
-    const phoneValidation = validatePhoneNumber(values.phoneNumber)
+
     const errors = Object.assign({}, validationResult.errors, {
-      phoneNumber: phoneValidation.error,
+      phoneNumber:
+        !values.phone.national || values.phone.isValid
```

Comment:
good question! `validator` on `Formik` component runs when the component is initially rendered. if the user doesn't have a phone number (which they don't when they add a new address instead of editing an address), we would populate `errors` object with a phone number error. 

to prevent that, we are _not_ creating an error if the phone number input is empty initially (if there's no `national` value), or phone number is validated (that happens only after they type a number). 

`Save` button on the form remains disabled until `isValid` is true. so the trick with `national` here is only for controlling when to render an error message.

Reply:
got it! thanks for the explanation :)